app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java

```diff
@@ -159,6 +171,24 @@ private void displayCoverImage(int position) {
         }
     }
 
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        if (orientation != newConfig.orientation) {
+            try {
+                orientation = newConfig.orientation;
+                getFragmentManager().beginTransaction().remove(this).commit();
+                getFragmentManager().beginTransaction()
+                        .replace(R.id.cover_fragment_container, CoverFragment.class.newInstance())
+                        .commitAllowingStateLoss();
```

Comment:
This looks a bit strange, to be honest. The fragment is shown in `R.id.cover_fragment_container` but that layout is located within the fragment that was just removed. Looks like this change is actually a lot harder than I thought... 

I have another idea how to do this. It is less hacky but it still is not really beautiful. The advantage of my version is that the images do not need to be re-loaded after orientation changes and that it does not do strange Fragment transactions but it is a bit ugly because it modifies the layout in Java code (sets LinearLayout orientation in `onConfigurationChanged`) and is less flexible. What do you think about that one?

https://github.com/ByteHamster/AntennaPod/commit/3c0f46ac865f566cd806b5857ae4c3a4c48816b8

Reply:
Thanks, I can fix weirdness with which id is removed 

Originally, I could have done what you suggested, but the look makes the image too small in vertical mode.

I will take a look again at my code so it is not strange

Reply:
I still think that this is pretty strange. What about just returning a `FrameLayout` in `onCreateView` and re-inflating its content in `onConfigurationChanged`?